aptos-move/e2e-move-tests/src/tests/fee_payer.rs

```diff
@@ -74,10 +90,169 @@ fn test_account_not_exist_with_fee_payer_create_account() {
+    let bob_after = h.read_aptos_balance(bob.address());
+
+    assert_eq!(h.sequence_number(alice.address()), 1);
+    assert!(bob_start > bob_after);
+}
+
+#[test]
+fn test_account_not_exist_out_of_gas_with_fee_payer() {
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = Account::new();
+    let beef = h.new_account_at(AccountAddress::from_hex_literal("0xbeef").unwrap());
+
+    // Load the code
+    assert_success!(h.publish_package_cache_building(
+        &beef,
+        &common::test_dir_path("infinite_loop.data/empty_loop"),
+    ));
+
+    let MemberId {
+        module_id,
+        member_id,
+    } = str::parse("0xbeef::test::run").unwrap();
+    let payload =
+        TransactionPayload::EntryFunction(EntryFunction::new(module_id, member_id, vec![], vec![]));
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(beef.clone())
+        .payload(payload)
+        .sequence_number(0)
+        .max_gas_amount(100_010) // This is the minimum to execute this transaction
+        .gas_unit_price(1)
+        .sign_fee_payer();
+    let result = h.run_raw(transaction);
+
+    assert_eq!(
+        result.status(),
+        &TransactionStatus::Keep(ExecutionStatus::MiscellaneousError(Some(
+            aptos_types::vm_status::StatusCode::EXECUTION_LIMIT_REACHED
+        ))),
+    );
+}
+
+#[test]
+fn test_account_not_exist_move_abort_with_fee_payer_out_of_gas() {
+    // Very large transaction to trigger the out of gas error aborted seqno 0 sponsored transactions
+    let mut h = MoveHarness::new_with_features(vec![FeatureFlag::GAS_PAYER_ENABLED], vec![]);
+
+    let alice = Account::new();
+    let bob = h.new_account_at(AccountAddress::from_hex_literal("0xb0b").unwrap());
+
+    let root = h.aptos_framework_account();
+
+    let data = vec![0; 1000 * 1024];
+    let entries = ApprovedExecutionHashes {
+        entries: vec![(0, HashValue::sha3_256_of(&data).to_vec())],
+    };
+
+    let script = Script::new(data, vec![], vec![]);
+
+    h.set_resource(
+        *root.address(),
+        ApprovedExecutionHashes::struct_tag(),
+        &entries,
+    );
+    let transaction = TransactionBuilder::new(alice.clone())
+        .fee_payer(bob.clone())
+        .script(script.clone())
+        .sequence_number(0)
+        .max_gas_amount(100_010) // This is the minimum to execute this transaction
+        .gas_unit_price(1)
+        .sign_fee_payer();
+    let result = h.run_raw(transaction);
+    assert_eq!(result.gas_used(), 100_010);
+
+    let new_alice = Account::new();
+    let transaction = TransactionBuilder::new(new_alice.clone())
+        .fee_payer(bob.clone())
+        .script(script.clone())
+        .sequence_number(0)
+        .max_gas_amount(100_011) // Bump by one to ensure more gas can be used
+        .gas_unit_price(1)
+        .sign_fee_payer();
+    let result = h.run_raw(transaction);
+    assert_eq!(result.gas_used(), 100_011);
```

Comment:
nit: if we trigger out of gas, would be nice to assert the returned status as well?

Reply:
these run out of gas in the epilogue, so we don't get those errors. only the logger would